client/src/js/services/receipts/modal/receiptModalController.js

```diff
@@ -10,31 +10,49 @@ ReceiptModalController.$inject = ['$uibModalInstance', '$window', 'receipt', 'op
  *                            by the service wrapping the receipt modal). The promise
  *                            is stored in an object to ensure the modal is evaluated
  *                            before the HTTP request (promise) is resolved.
- * @param {String} template  Path to the template or resource to load
- * @param {String} render    Render target used to generate report
+ * @params {String} template  Path to the template or resource to load
+ * @params {String} render    Render target used to generate report
  */
-function ReceiptModalController($modalInstance, $window, receipt, options) {
+function ReceiptModalController($modalInstance, $window, $sce, Notify, receipt, options) {
   var vm = this;
-
+  
+  vm.print = print;
+  vm.close = close;
+  
   // expose options to the view
   angular.extend(vm, options);
 
   receipt.promise
     .then(function (result) {
-
-      // receipt has successfully loaded
-      vm.receipt = result;
+      // special case for pdf rendering
+      if (options.renderer === 'pdf') { 
+        // store downloaded base64 PDF file in a browser blob - this will be accessible through 'blob://...'
+        var file = new Blob([result], {type : 'application/pdf'});
+        
+        // determine the direct path to the newly (temporarily) stored PDF file
+        var fileURL = URL.createObjectURL(file);
+  
+        // trust and expose the file to the view to embed the PDF
+        vm.receipt = $sce.trustAsResourceUrl(fileURL);  
+      } else { 
+        // simply expose receipt object to view
+        vm.receipt = result;
+      }
     })
-    .catch(function (error) {
-
-      // receipt has failed - show error state
-    });
-
-  vm.print = function print() {
+    .catch(Notify.handleError);
+  
+  function print() {
+    /**@todo This printing could be exposed by a directive/ component */
+    // special case for embedded content
+    if (options.renderer === 'pdf') {
```

Comment:
There are several places where this check occurs.  I wonder if it might be better to have a constant (or service) like `renderers.PDF` that these checks could use so that it isn't a hardcoded string all the time.  something like:
```js
if (options.renderer === renderers.PDF) {
  // logic
}
```